elasticgraph-graphql/lib/elastic_graph/graphql/filtering/filter_interpreter.rb

```diff
@@ -103,6 +103,8 @@ def filters_on_sub_fields?(expression)
         end
 
         def process_not_expression(bool_node, expression, field_path)
+          return if expression.nil? || expression == {}
```

Comment:
According to the RBS signature, `expression` cannot be `nil`. Before now that was true but with your refactoring it can now be `nil`.

Can you update the RBS signatures for these methods to account for the new possible values?

Reply:
Done